plugins/modules/ipa_pwpolicy.py

```diff
@@ -159,26 +185,35 @@ def pwpolicy_del(self, name):
 
 def get_pwpolicy_dict(maxpwdlife=None, minpwdlife=None, historylength=None, minclasses=None,
                       minlength=None, priority=None, maxfailcount=None, failinterval=None,
-                      lockouttime=None):
+                      lockouttime=None, gracelimit=None, maxrepeat=None, maxsequence=None, dictcheck=None, usercheck=None):
     pwpolicy = {}
-    if maxpwdlife is not None:
-        pwpolicy['krbmaxpwdlife'] = maxpwdlife
-    if minpwdlife is not None:
-        pwpolicy['krbminpwdlife'] = minpwdlife
-    if historylength is not None:
-        pwpolicy['krbpwdhistorylength'] = historylength
-    if minclasses is not None:
-        pwpolicy['krbpwdmindiffchars'] = minclasses
-    if minlength is not None:
-        pwpolicy['krbpwdminlength'] = minlength
-    if priority is not None:
-        pwpolicy['cospriority'] = priority
-    if maxfailcount is not None:
-        pwpolicy['krbpwdmaxfailure'] = maxfailcount
-    if failinterval is not None:
-        pwpolicy['krbpwdfailurecountinterval'] = failinterval
-    if lockouttime is not None:
-        pwpolicy['krbpwdlockoutduration'] = lockouttime
+    pwpolicy_options = {
+        'krbmaxpwdlife': maxpwdlife,
+        'krbminpwdlife': minpwdlife,
+        'krbpwdhistorylength': historylength,
+        'krbpwdmindiffchars': minclasses,
+        'krbpwdminlength': minlength,
+        'cospriority': priority,
+        'krbpwdmaxfailure': maxfailcount,
+        'krbpwdfailurecountinterval': failinterval,
+        'krbpwdlockoutduration': lockouttime,
+        'passwordgracelimit': gracelimit,
+        'ipapwdmaxrepeat': maxrepeat,
+        'ipapwdmaxsequence': maxsequence,
+    }
+
+    pwpolicy_boolean_options = {
+        'ipapwddictcheck': dictcheck,
+        'ipapwdusercheck': usercheck,
+    }
+
+    for option, value in pwpolicy_options.items():
+        if value is not None:
+            pwpolicy[option] = str(value)
```

Comment:
It's better to use `to_native` to avoid problems converting `unicode` strings (in Python 2.7) to `str`:
```suggestion
            pwpolicy[option] = to_native(value)
```

Reply:
Done